src/firebase/firebase.js

```diff
@@ -19,6 +20,40 @@ class FireBase {
     };
     firebase.initializeApp(config);
   }
+
+  static async auth() {
+    const token = await firebase.auth().signInWithPopup(new firebase.auth.GithubAuthProvider())
+      .then(result => result.credential.accessToken)
+      .catch((error) => {
+        throw new Error(error);
+      });
+
+    if (token) {
+      const userData = await fetch(`https://api.github.com/user?access_token=${token}`)
+        .then(response => response.json())
+        .catch((error) => {
+          throw new Error(error);
+        });
+
+      const user = firebase.auth().currentUser;
+
+      user.updateProfile({
+        displayName: userData.login,
+        photoURL: userData.avatar_url,
+      }).catch((error) => {
+        throw new Error(error);
+      });
+    }
+  }
+
+  static logout() {
+    firebase.auth().signOut().then(() => {
+      window.location.reload();
+    }).catch((error) => {
+      throw new Error(error);
```

Comment:
it is better to create a normal error  and add console.error('logout error', e);